src/test/java/com/answerdigital/answerking/service/ProductServiceTest.java

```diff
@@ -30,154 +30,162 @@
 import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
-class ProductServiceTest {
+final class ProductServiceTest {
+
+    @InjectMocks
+    private ProductService productService;
 
     @Mock
     private ProductRepository productRepository;
 
     @Mock
     private CategoryService categoryService;
 
-    @InjectMocks
-    private ProductService productService;
-
-    private Product product;
+    private final ProductTestBuilder productTestBuilder;
 
-    private Category category;
+    private final ProductRequestTestBuilder productRequestTestBuilder;
 
-    private ProductRequest productRequest;
+    private final CategoryTestBuilder categoryTestBuilder;
 
-    private static final long PRODUCT_ID = 55L;
+    private static final long NONEXISTENT_PRODUCT_ID = 10L;
 
-    @BeforeEach
-    public void generateProduct() {
-        category = Category.builder()
-                .name("test")
-                .description("categoryDesc")
-                .id(1L)
-                .products(new HashSet<>())
-                .build();
-        product = Product.builder()
-                .id(PRODUCT_ID)
-                .name("test")
-                .description("testDes")
-                .price(BigDecimal.valueOf(2.99))
-                .retired(false)
-                .category(category)
-                .build();
-        productRequest = ProductRequest.builder()
-                .name("test")
-                .description("testD")
-                .price(BigDecimal.valueOf(1.99))
-                .categoryId(1L)
-                .build();
+    private ProductServiceTest() {
+        productTestBuilder = new ProductTestBuilder();
+        productRequestTestBuilder = new ProductRequestTestBuilder();
+        categoryTestBuilder = new CategoryTestBuilder();
     }
 
     @Test
     void addNewProductReturnsProductObjectSuccessfully() {
-        //given
-        when(productRepository.save(any())).thenReturn(product);
+        // given
+        final Product expectedProduct = productTestBuilder.withDefaultValues().build();
+        final ProductRequest productRequest = productRequestTestBuilder.withDefaultValues().build();
+        final Category category = categoryTestBuilder.withDefaultValues().build();
+
+        // when
+        when(productRepository.save(any())).thenReturn(expectedProduct);
```

Comment:
Could we be explicit with the any method, for example:

```suggestion
        when(productRepository.save(any(Product.class))).thenReturn(expectedProduct);
```

not sure if it's really necessary, was thinking more long the lines of readability 😄 also reflect throughout!